renderer/components/GamepadList.tsx

```diff
@@ -1,17 +1,21 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { Panel, ListGroup } from 'react-bootstrap';
 import { connect } from 'react-redux';
-
 import _ from 'lodash';
-import Gamepad from './Gamepad';
+import { Gamepad, GamepadObject } from './Gamepad';
+
+interface StateProps {
+  gamepads: object;
+}
```

Comment:
`gamepads` is an object that contains multiple `GamepadObject` (I think). Might be helpful to be more specific than just leaving it as an object.

Reply:
By multiple `GamepadObject` do you mean a list of them? If so, I think you can do `GamepadObject[]`

Reply:
In the proptype assertion, it said that `gamepads` is an object.

Reply:
I see. We seem to have a type mismatch. In the state, `state.gamepads.gamepads` is `string | undefined`. But yes from the name I would think it's an array of gamepad objects, also because of the `.map` too.

Reply:
That's a little strange. Do you have any idea why there would be a mismatch? Not sure whether if it's the component or the actions/reducer that is the problem here.

Reply:
Hmm yea, from a brief look, I think we might have specified the wrong type in the actions/reducer.